backup/server_stats.py

```diff
@@ -9,11 +9,14 @@ def within_tolerance(current, previous):
     return relative_size_change(current, previous) <= allowed_size_change(previous)
 
 
-def date_to_prefix(prefix, date):
+def date_to_prefix(prefix, date, file_date_format):
     '''Converts a date to a S3 prefix of the form: prefix/year.month.day.
     Month and day are prepended with a leading zero if it is only 1 letter long.
     '''
-    return '/'.join([prefix, f'{date.year}.{date.month:02}.{date.day:02}'])
+    if file_date_format:
+        return '/'.join([prefix, date.strftime(file_date_format)])
+    else:
+        return '/'.join([prefix, f'{date.year}.{date.month:02}.{date.day:02}'])
```

Comment:
Can you also make sure we support it if the prefix is an empty string?
I think we'll need that in the logDNA use case

Reply:
Maybe this function would also be helpful: https://docs.python.org/3.6/library/os.path.html#os.path.join

Reply:
The prefix input for this method is the folder argument u pass into the main. So if you make sure and empty string is put in for that argument, then this should work as expected.

Also, not sure to use os.path.join, since that builds an OS specific path and we are trying to build a path for the s3 bucket..

Reply:
>  So if you make sure and empty string is put in for that argument, then this should work as expected.
I would prefer paths not to consist of double slashes because the prefix was an empty string. That's what I was trying to get at. With `'/'.join(['bucketx', '', 'bla'])` you would get the output `bucketx//bla`
> Also, not sure to use os.path.join, since that builds an OS specific path and we are trying to build a path for the s3 bucket..
`os.sep` defaults to `/` on posix systems. Also you can provide it as an argument if you really like.

I just figured that if the `file_date_format` is empty you can just overwrite it with a date format that would result in `f'{date.year}.{date.month:02}.{date.day:02}'`

Reply:
Refactored it to take these comments in account.